lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -1243,21 +1244,28 @@ bool swift::getAllBorrowIntroducingValues(SILValue inputValue,
     // instruction
     if (isGuaranteedForwarding(value)) {
       if (auto *i = value->getDefiningInstruction()) {
-        llvm::copy(i->getNonTypeDependentOperandValues(),
-                   std::back_inserter(worklist));
+        for (SILValue opValue : i->getNonTypeDependentOperandValues()) {
+          worklist.insert(opValue);
+        }
         continue;
       }
 
       // Otherwise, we should have a block argument that is defined by a single
       // predecessor terminator.
       auto *arg = cast<SILPhiArgument>(value);
-      auto *termInst = arg->getSingleTerminator();
-      assert(termInst && termInst->isTransformationTerminator() &&
-             OwnershipForwardingMixin::get(termInst)->preservesOwnership());
-      assert(termInst->getNumOperands() == 1 &&
-             "Transforming terminators should always have a single operand");
-      worklist.push_back(termInst->getAllOperands()[0].get());
-      continue;
+      if (!arg->isPhi()) {
+        auto *termInst = arg->getSingleTerminator();
+        assert(termInst && termInst->isTransformationTerminator() &&
+               OwnershipForwardingMixin::get(termInst)->preservesOwnership());
+        assert(termInst->getNumOperands() == 1 &&
+               "Transforming terminators should always have a single operand");
+        worklist.insert(termInst->getAllOperands()[0].get());
```

Comment:
Instead of all this you can do:

```
if (auto *result = SILArgument::isTerminatorResult()) 
  if (auto *forwardedOper = result->forwardedTerminatorResultOperand()) ...
```
At least once this lands.
https://github.com/apple/swift/pull/62493
should I merge it now?

In the future, it should be something like:

```
if (auto result = TerminatorResult(value)) {
  if (auto *forwardedOper = result.forwardedOper()) ...
}